app/src/main/java/com/jetpack/compose/learning/canvas/CanvasDrawingBoardActivity.kt

```diff
@@ -0,0 +1,321 @@
+                                    motionEvent = MotionEvent.Up
+                                    if (pointer.pressed != pointer.previousPressed) pointer.consume()
+                                }
+                            }
+                        }
+                    }
+            ) {
+                // Draw here
+                when (motionEvent) {
+                    MotionEvent.Down -> {
+                        currentPath.moveTo(currentPosition.x, currentPosition.y)
+                        previousPosition = currentPosition
+                    }
+                    MotionEvent.Move -> {
+                        currentPath.quadraticBezierTo(
+                            previousPosition.x,
+                            previousPosition.y,
+                            (previousPosition.x + currentPosition.x) / 2,
+                            (previousPosition.y + currentPosition.y) / 2
+
+                        )
+                        previousPosition = currentPosition
+                    }
+                    MotionEvent.Up -> {
+                        // Save path and rest all properties
+                        currentPath.lineTo(currentPosition.x, currentPosition.y)
+                        paths.add(Pair(currentPath, currentPathProperty))
+                        currentPath = Path()
+                        currentPathProperty = PathCharacteristics(
+                            strokeWidth = currentPathProperty.strokeWidth,
+                            color = currentPathProperty.color
+                        )
+                        currentPosition = Offset.Unspecified
+                        previousPosition = currentPosition
+                        motionEvent = MotionEvent.Idle
+                    }
+                    else -> Unit
+                }
+
+                // draw all the paths with individual stroke and color
+                paths.forEach {
+                    val path = it.first
+                    val property = it.second
+                    drawPath(
+                        color = property.color,
+                        path = path,
+                        style = Stroke(
+                            width = property.strokeWidth,
+                            cap = StrokeCap.Round, join = StrokeJoin.Round
+                        )
+                    )
+                }
+
+                if (motionEvent != MotionEvent.Idle) {
+                    // draw path based on current ongoing gesture
+                    drawPath(
+                        color = currentPathProperty.color,
+                        path = currentPath,
+                        style = Stroke(
+                            width = currentPathProperty.strokeWidth,
+                            cap = StrokeCap.Round, join = StrokeJoin.Round
+                        )
+                    )
+                }
+            }
+        }
+    }
+
+    enum class MotionEvent {
+        Idle, Down, Move, Up
```

Comment:
Since they are constants, the names of an enum type's fields should be in uppercase letters.